reach

```diff
@@ -115,10 +115,103 @@ run_s () {
   sh -ac "$TMP/out.sh" "$0"
 }
 
+package_manager () {
```

Comment:
I'm not sure on this one though. I'm considering that some users doesn't know what a package manager is. It would be good if it could choose automatically to those users.

Let me know your thoughts.

Reply:
Yup, that's what I'm saying; `pkg_i` would still detect their package manager but instead of multiple functions that all do essentially the same thing and using `export` to coordinate between them, you'd just consolidate into one function that takes `make|curl|whatever` as its single argument.

Users still won't need to understand how it works.

Reply:
Got it. I did some changes there.

Thanks :)

Reply:
Again, please re-read my comments above - you're still `export`ing a `PACKAGE_MANAGER` unnecessarily.

Reply:
> Yup, that's what I'm saying; pkg_i would still detect their package manager but instead of multiple functions that all do essentially the same thing and using export to coordinate between them, you'd just consolidate into one function that takes make|curl|whatever as its single argument.

You've removed the `export`s but still have multiple functions instead of just `package_install`. I'm not sure how this could work right now since you're not actually sharing `apt` or `yum` between them.

Reply:
I have merged the package_manager function into package_install. It was not being used anywhere else but on package_install. Good catch!